src/vcpkg/metrics.cpp

```diff
@@ -608,32 +518,98 @@ namespace vcpkg
         fs.write_contents(vcpkg_metrics_txt_path, payload, ec);
         if (ec) return;
 
-#if defined(_WIN32)
-        const Path temp_folder_path_exe = temp_folder_path / "vcpkg-" VCPKG_BASE_VERSION_AS_STRING ".exe";
+        const Path temp_folder_path_exe = temp_folder_path / "vcpkg-" VCPKG_BASE_VERSION_AS_STRING
+#if defined(WIN32)
+                                                             ".exe"
+#endif
+            ;
         fs.copy_file(get_exe_path_of_current_process(), temp_folder_path_exe, CopyOptions::skip_existing, ec);
         if (ec) return;
+
         Command builder;
         builder.string_arg(temp_folder_path_exe);
         builder.string_arg("z-upload-metrics");
         builder.string_arg(vcpkg_metrics_txt_path);
         cmd_execute_background(builder);
-#else
-        cmd_execute_background(Command("curl")
-                                   .string_arg("https://dc.services.visualstudio.com/v2/track")
-                                   .string_arg("--max-time")
-                                   .string_arg("60")
-                                   .string_arg("-H")
-                                   .string_arg("Content-Type: application/json")
-                                   .string_arg("-X")
-                                   .string_arg("POST")
-                                   .string_arg("--tlsv1.2")
-                                   .string_arg("--data")
-                                   .string_arg(Strings::concat("@", vcpkg_metrics_txt_path))
-                                   .raw_arg(">/dev/null")
-                                   .raw_arg("2>&1")
-                                   .raw_arg(";")
-                                   .string_arg("rm")
-                                   .string_arg(vcpkg_metrics_txt_path));
-#endif
+    }
+
+    static size_t string_append_cb(void* buff, size_t size, size_t nmemb, void* param)
+    {
+        auto* str = reinterpret_cast<std::string*>(param);
+        if (!str || !buff) return 0;
+        if (size != 1) return 0;
+        str->append(reinterpret_cast<char*>(buff), nmemb);
+        return size * nmemb;
+    }
+
+    bool parse_metrics_response(StringView response_body)
+    {
+        auto maybe_json = Json::parse_object(response_body, "metrics_response");
+        auto json = maybe_json.get();
+        if (!json) return false;
+
+        auto maybe_received = json->get(vcpkg::AppInsightsResponseItemsReceived);
+        auto maybe_accepted = json->get(vcpkg::AppInsightsResponseItemsAccepted);
+        auto maybe_errors = json->get(vcpkg::AppInsightsResponseErrors);
+
+        if (maybe_received && maybe_accepted && maybe_errors && maybe_received->is_integer() &&
+            maybe_accepted->is_integer() && maybe_errors->is_array())
+        {
+            auto item_received = maybe_received->integer(VCPKG_LINE_INFO);
+            auto item_accepted = maybe_accepted->integer(VCPKG_LINE_INFO);
```

Comment:
Not a fan of this check followed by assert pattern but there's currently no `maybe_integer` so no change requested for now.